### DISCUSSION TITLE
4.x Window nodes need Scale, rotation and Pivot offsets

### DISCUSSION THREAD

**Discussion by ondesic** [godotengine/godot-proposals#11463] [2D]
Moving a project from Godot 3.6 to 4.3/4.4dev7.
My project relies heavily on different ConfirmationDialogs/AcceptDialogs and Windows for their exclusive nature and interaction. 
The structure of Window Nodes are drastically different in 4.x and, in some ways, not usable compared to 3.6.

In my project each WindowDialog opens with nice Scaling animation around the Pivot offset of the Window. Some animation rotates open or closed.
Windows now are missing overall scale, rotation and pivotoffset like other controls have. Now a window cannot be animated to size (0,0) like scaling allows. Furthermore, the loss of a Pivot point will not allow for rotation either, or scaling from the center effect.

Can this be added back?

**Comment by sockeye-d** [comment]
I don't think it can since in gd3, Windows were forced to be embedded but now in gd4 individual Window nodes will appear as separate windows (if embed windows is turned off in the project settings)

**Comment by ondesic** [comment]
How can we animate Windows/AcceptDialogs etc? This is important for professional look

**Reply by sockeye-d** [reply]
Don't use windows, create your own solution for a container you can drag around by the top. The OS's window manager animates windows opening and closing already, and I would assume most don't even support rotated windows. If it's important that the floating window can display over other windows, you can create a new Window node and set it to fullscreen and transparent, turn on mouse passthrough, and set the mouse passthrough polygon to the bounds of your custom window container
